File: core/src/SprCSharp/SprCSharp/tmp/arraywrapper_PHOpParticleDesc.cs
    public class arraywrapper_PHOpParticleDesc : arraywrapper {
        public arraywrapper_PHOpParticleDesc(int nelm) : base(nelm) {
            _this = SprExport.Spr_array_new_PHOpParticleDesc(nelm);
            _flag = true;
        }
        public arraywrapper_PHOpParticleDesc(IntPtr ptr, int nelm) : base(ptr, nelm) {}
        public arraywrapper_PHOpParticleDesc(IntPtr ptr, int nelm, bool flag) : base(ptr, nelm, flag) {}
        ~arraywrapper_PHOpParticleDesc() {
            if (_flag) { SprExport.Spr_array_delete_PHOpParticleDesc(_this); _flag = false; }
        }
        public PHOpParticleDesc this[int index] {
            get { return new PHOpParticleDesc(SprExport.Spr_array_get_PHOpParticleDesc(_this, index)); }
            set { SprExport.Spr_array_set_PHOpParticleDesc(_this, index, value); }
        }
        public int size() { return (int) _nelm; }
        public static implicit operator List<PHOpParticleDesc>(arraywrapper_PHOpParticleDesc m) {
            List<PHOpParticleDesc> r = new List<PHOpParticleDesc>();
            for (int i = 0; i < (int) m._nelm; i++) {
                IntPtr member = SprExport.Spr_array_get_PHOpParticleDesc(m._this, i);
                r.Add(new PHOpParticleDesc(member));
            }
            return r;
        }
        public static implicit operator arraywrapper_PHOpParticleDesc(List<PHOpParticleDesc> r) {
            int size = r.Count;
            IntPtr ptr = SprExport.Spr_array_new_PHOpParticleDesc(size);
            arraywrapper_PHOpParticleDesc m = new arraywrapper_PHOpParticleDesc(ptr, size, true);
            for (int i = 0; i < size; i++) {
                SprExport.Spr_array_set_PHOpParticleDesc(m._this, i, r[i]);
            }
            return m;
        }
        public static implicit operator PHOpParticleDesc[](arraywrapper_PHOpParticleDesc m) {
            int size = (int) m._nelm;
            PHOpParticleDesc[] r = new PHOpParticleDesc[size];
            for (int i = 0; i < size; i++) {
                 IntPtr ptr = SprExport.Spr_array_get_PHOpParticleDesc(m._this, i);
                 r[i] = new PHOpParticleDesc(ptr);
            }
            return r;
        }
        public static implicit operator PHOpParticleDesc(arraywrapper_PHOpParticleDesc m) {
            int size = (int) m._nelm;
            IntPtr ptr = SprExport.Spr_array_get_PHOpParticleDesc(m._this, 0);
            PHOpParticleDesc r = new PHOpParticleDesc(ptr);
            return r;
        }
        public static implicit operator arraywrapper_PHOpParticleDesc(PHOpParticleDesc[] r) {
            int size = r.Length;
            IntPtr ptr = SprExport.Spr_array_new_PHOpParticleDesc(size);
            arraywrapper_PHOpParticleDesc m = new arraywrapper_PHOpParticleDesc(ptr, size, true);
            for (int i = 0; i < size; i++) {
                SprExport.Spr_array_set_PHOpParticleDesc(m._this, i, r[i]);
            }
            return m;
        }
    }

File: core/src/SprCSharp/SprExport/tmp/arraywrapper_PHOpParticleDesc.cpp
    __declspec(dllexport) HANDLE __cdecl Spr_array_new_PHOpParticleDesc(int size) {
        HANDLE _ptr = NULL;
        try { _ptr = new PHOpParticleDesc[size]; }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
        return _ptr;
    }
    __declspec(dllexport) void __cdecl Spr_array_delete_PHOpParticleDesc(HANDLE ptr) {
        if (ptr == NULL) return;
        try { delete[] (PHOpParticleDesc*) ptr; }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
    }
    __declspec(dllexport) HANDLE __cdecl Spr_array_get_PHOpParticleDesc(HANDLE ptr, int index) {
        HANDLE _ptr = NULL;
        try {
            PHOpParticleDesc* aryptr = (PHOpParticleDesc*) ptr;
            _ptr = (HANDLE) &aryptr[index];
        }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
        return _ptr;
    }
    __declspec(dllexport) void __cdecl Spr_array_set_PHOpParticleDesc(HANDLE ptr, int index, HANDLE value) {
        try {
            PHOpParticleDesc* aryptr = (PHOpParticleDesc*) ptr;
            aryptr[index] = *((PHOpParticleDesc*) value);
        }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
    }
